indra/llui/lltextbase.cpp

```diff
@@ -3801,6 +3801,21 @@ LLEmojiTextSegment::LLEmojiTextSegment(const LLUIColor& color, S32 start, S32 en
 {
 }
 
+
+F32 LLEmojiTextSegment::draw(S32 start, S32 end, S32 selection_start, S32 selection_end, const LLRectf& draw_rect)
+{
+    bool reset_font_buffers = (mLastGeneration != mEditor.getTextGeneration()) && (mLastGeneration != -1);
+
+    F32 result = LLNormalTextSegment::draw(start, end, selection_start, selection_end, draw_rect);
```

Comment:
Approved, but this seems odd. Why does it need to wait for other segments? Buffer is a collection of points, it shouldn't need to reset unless position changes because points remain unchanged.